mlpf/pyg/mlpf.py

```diff
@@ -179,4 +180,13 @@ def forward(self, event):
         preds_momentum = torch.cat([preds_pt, preds_eta, preds_phi, preds_energy], axis=-1)
         pred_charge = self.nn_charge(embedding_reg)
 
-        return preds_id, preds_momentum, pred_charge
+        ypred = unpack_predictions(preds_id, preds_momentum, pred_charge)
+
+        # must return the ygen (and ycand) too for torch_geometric.nn.data_parallel
+        ygen = unpack_target(event.ygen)
+        try:
+            ycand = unpack_target(event.ycand)
+        except Exception:
+            ycand = None
+
+        return ygen, ycand, ypred
```

Comment:
In my opinion, the forward pass of the model should take only `event.X` as input and only return `ypred`. Targets shouldn't be unpacked in the forward pass of the model since we want to be able to run the model without the overhead of unpacking targets and in principle even without having access to targets.